### ISSUE TITLE
Supports arbitrarily changing the order of different cases.

### ISSUE THREAD

**Issue by janryWang** [evanwashere/mitata#18]
Because as long as the benchmark is executed by a single process, it will basically face the interaction between different use cases. Changing the order will have a great impact on the final data. To solve this problem, we can only solve it with the help of sub-processes, just like [iso-bench](https://github.com/Llorx/iso-bench), but The problem is how to combine it with d8. This is another very troublesome problem.

**Comment by evanwashere** [comment]
this is only true for functions that get deoptimized too often or don't meet jit rules and are stuck in lower tiers forever

if changing the order of functions changes the result then it means you are hiding something from jit and hinting it to make (de-)optimizations that it wouldn't otherwise

real codebases are full of these bad/good hints and you should not tailor your benchmarks to some magic function order that makes it 5x faster for no apparent reason

(next version will remove first benchmark jit bias problem if that's what you are concerned about)

**Comment by janryWang** [comment]
The problem is that I just wrote such a simple example and adjusted the order slightly, and the results were very different. This made me seriously doubt the accuracy of mitata's data. I didn't even know what order I should use. Writing use cases can ensure data accuracy and stability
```ts
import { run, bench, group } from 'mitata'

class A {}

class B {
  aa: any
  bb: any
  constructor(aa?: any, bb?: any) {
    this.aa = aa ?? this
    this.bb = bb ?? 'value'
  }
}
group('basic', () => {
 // change order
  bench('noop', () => {})
  bench('class A', () => {
    void new A()
  })
  bench('class B', () => {
    void new B()
  })
})

run()

```
![image](https://github.com/evanwashere/mitata/assets/4060976/a0d33cff-de43-4498-b02a-0883727cfa98)
![image](https://github.com/evanwashere/mitata/assets/4060976/dc52d130-d20c-4a35-80a1-02d9f3e85a29)
![image](https://github.com/evanwashere/mitata/assets/4060976/6aed66b7-c1e3-435f-ad0a-cf83a7a53425)

**Comment by gwsbhqt** [comment]
I also encountered the same problem.

**Comment by evanwashere** [comment]
> @evanwashere The problem is that I just wrote such a simple example and adjusted the order slightly, and the results were very different. This made me seriously doubt the accuracy of mitata's data. I didn't even know what order I should use. Writing use cases can ensure data accuracy and stability
> 
> ```ts
> 
> import { run, bench, group } from 'mitata'
> 
> 
> 
> class A {}
> 
> 
> 
> class B {
> 
>   aa: any
> 
>   bb: any
> 
>   constructor(aa?: any, bb?: any) {
> 
>     this.aa = aa ?? this
> 
>     this.bb = bb ?? 'value'
> 
>   }
> 
> }
> 
> group('basic', () => {
> 
>  // change order
> 
>   bench('noop', () => {})
> 
>   bench('class A', () => {
> 
>     void new A()
> 
>   })
> 
>   bench('class B', () => {
> 
>     void new B()
> 
>   })
> 
> })
> 
> 
> 
> run()
> 
> 
> 
> ```
> 
> ![image](https://github.com/evanwashere/mitata/assets/4060976/a0d33cff-de43-4498-b02a-0883727cfa98)
> 
> ![image](https://github.com/evanwashere/mitata/assets/4060976/dc52d130-d20c-4a35-80a1-02d9f3e85a29)
> 
> ![image](https://github.com/evanwashere/mitata/assets/4060976/6aed66b7-c1e3-435f-ad0a-cf83a7a53425)
> 
> 
> 
> 


what you are observing is jit bias. mitata has tight near-zero overhead loop, hinting the jit to fully inline the first function passed to `bench()` and specialize optimizations further on that.
However, in the next benchmark, the jit sees that `bench()` is now taking a completely different function and undoes all inlining optimizations, with no further (or limited) attempts to reapply them. This results in a ~4ns baseline for following bench() calls

While this ~4ns fixed cost is mostly noise for comparisons exceeding ~50ns, I'm working on removing it and making sure every function gets same inline treatment across v8, jsc and other engines

**Comment by evanwashere** [comment]
fixed in mitata@0.1.7

**Comment by gwsbhqt** [comment]
@evanwashere nice job!
